packages/assets-controllers/src/DeFiPositionsController/DeFiPositionsController.ts

```diff
@@ -0,0 +1,165 @@
+  typeof controllerName,
+  DeFiPositionsControllerState
+>;
+
+export type DeFiPositionsControllerEvents =
+  DeFiPositionsControllerStateChangeEvent;
+
+export type DeFiPositionsControllerStateChangeEvent =
+  ControllerStateChangeEvent<
+    typeof controllerName,
+    DeFiPositionsControllerState
+  >;
+
+/**
+ * The external actions available to the {@link DeFiPositionsController}.
+ */
+export type AllowedActions =
+  | AccountsControllerGetAccountAction
+  | AccountsControllerGetSelectedAccountAction;
+
+/**
+ * The external events available to the {@link DeFiPositionsController}.
+ */
+export type AllowedEvents =
+  | NetworkControllerStateChangeEvent
+  | AccountsControllerSelectedAccountChangeEvent;
+
+/**
+ * The messenger of the {@link DeFiPositionsController}.
+ */
+export type DeFiPositionsControllerMessenger = RestrictedMessenger<
+  typeof controllerName,
+  DeFiPositionsControllerActions | AllowedActions,
+  DeFiPositionsControllerEvents | AllowedEvents,
+  AllowedActions['type'],
+  AllowedEvents['type']
+>;
+
+/**
+ * Controller that stores assets and exposes convenience methods
+ */
+export class DeFiPositionsController extends BaseController<
+  typeof controllerName,
+  DeFiPositionsControllerState,
+  DeFiPositionsControllerMessenger
+> {
+  // TODO: Confirm whether we can store the account address instead of the id
+  // Storing the address means we don't need to query it in every event handler
+  #selectedAccountId: string;
+
+  /**
+   * Tokens controller options
+   *
+   * @param options - Constructor options.
+   * @param options.messenger - The controller messenger.
+   * @param options.state - Initial state to set on this controller.
+   */
+  constructor({
+    messenger,
+    state,
+  }: {
+    messenger: DeFiPositionsControllerMessenger;
+    state?: Partial<DeFiPositionsControllerState>;
+  }) {
+    super({
+      name: controllerName,
+      metadata: controllerMetadata,
+      messenger,
+      state: {
+        ...getDefaultDefiPositionsControllerState(),
+        ...state,
+      },
+    });
+
+    this.#selectedAccountId = this.messagingSystem.call(
+      'AccountsController:getSelectedAccount',
+    ).id;
+
+    this.messagingSystem.subscribe(
+      'AccountsController:selectedAccountChange',
+      async (selectedAccount) => {
+        this.#selectedAccountId = selectedAccount.id;
+
+        await this.#updateAccountPositions(selectedAccount.address);
+      },
+    );
+
+    this.messagingSystem.subscribe(
+      'NetworkController:stateChange',
+      async () => {
+        const selectedAddress = this.messagingSystem.call(
+          'AccountsController:getAccount',
+          this.#selectedAccountId,
+        );
+
+        if (selectedAddress) {
+          await this.#updateAccountPositions(selectedAddress.address);
+        }
+      },
+    );
+  }
+
+  // TODO: If this becomes an action, accountAddress needs to be inferred from the id
+  async #updateAccountPositions(accountAddress: string) {
```

Comment:
Eventually, we might expose this as an action to refetch positions from UI. I have added a comment so that when we do so, we make sure the `accountAddress` is not required and we instead fetch the address from the `#selectedAccountId`.